.github/workflows/crictl/Dockerfile-centos8.1

```diff
@@ -0,0 +1,93 @@
+#stage 1: build crictl-src
```

Comment:
Why does CI/CD need to muti-build the Dockerfile for each pull request? I think just pulling the docker images from Dockerhub is OK.

Reply:
First, It is not building an image for each pull request. This is a manual trigger. We will build the image each month based on occlum image. Stage 1 building is to obtain the circtl environment binary and configure file and copy to occlum image.
Second, the image should be able to compile occlum, so we will build a new crictl-test image based on occlum image.
Third, the build process is based on Github machine rather than a self-hosted machine, the build process is quick. About 15 minutes will complete the build image and push it to docker hub.
I don't understand why just pulling the docker images from Dockerhub is OK. What images can I pull to be able to compile occlum and with cricitl environment?

Reply:
> First, It is not building an image for each pull request. This is a manual trigger. We will build the image each month based on occlum image. Stage 1 building is to obtain the circtl environment binary and configure file and copy to occlum image.
> Second, the image should be able to compile occlum, so we will build a new crictl-test image based on occlum image.
> Third, the build process is based on Github machine rather than a self-hosted machine, the build process is quick. About 15 minutes will complete the build image and push it to docker hub.
> I don't understand why just pulling the docker images from Dockerhub is OK. What images can I pull to be able to compile occlum and with cricitl environment?

OK,  So:
1. You should split this `dockerfile` into another commit and write your reasons in the corresponding commit head.
2. Split stage 1 into another `dockerfile` because it it a constant images and push it to dockerhub
3. Change this `dockerfile` like this:
```Dockerfile
#stage 1: pull crictl-src
FROM crictl-src AS crictl-src
# stage 2: build crictl-test
FROM occlum/occlum:OCCLUM_VERSION-centos8.1 AS crictl-test

LABEL maintainer="Shirong Hao <shirong@linux.alibaba.com>"

RUN yum install -y libseccomp-devel iptables openssl openssl-devel wget make

WORKDIR /root

RUN mkdir -p /usr/local/go                                  \
    /etc/containerd                                         \
    /etc/cni/net.d                                          \
    /opt/cni                                                \
    /root/samples                                           \
    /tmp/hello_world

# copy binary and configure files
COPY --from=crictl-src /opt/cni                             /opt/cni
COPY --from=crictl-src /etc/containerd/config.toml          /etc/containerd
COPY --from=crictl-src /usr/local/bin                       /usr/local/bin

COPY 10-mynet.conf                                          /etc/cni/net.d
COPY 99-loopback.conf                                       /etc/cni/net.d
COPY crictl.yaml                                            /etc
COPY samples                                                /root/samples
COPY Dockerfile                                             /root
COPY hello_world.c                                          /root

# install docker-ce
RUN yum install -y yum-utils device-mapper-persistent-data lvm2 && \
    yum-config-manager --add-repo http://mirrors.aliyun.com/docker-ce/linux/centos/docker-ce.repo && \
    yum -y install docker-ce-18.03.0.ce

# configure docker
RUN mkdir -p /etc/docker && \
    echo -e "{\n\t\"runtimes\": {\n\t\t\"rune\": {\n\t\t\t\"path\": \"/usr/local/bin/rune\",\n\t\t\t\"runtimeArgs\": []\n\t\t}\n\t}\n}" >> /etc/docker/daemon.json
```

Reply:
After discussion, we think it is unnecessary to save the first stage image to Docker Hub. The reason is that no need to maintain an intermediate mirror since the time and resource cost is completely acceptable.